src/compiler/checker.ts

```diff
@@ -981,6 +982,29 @@ namespace ts {
             }
         }
 
+        function checkAndReportErrorForUsingTypeAsValue(errorLocation: Node, name: string, meaning: SymbolFlags, nameNotFoundMessage: DiagnosticMessage, nameArg: string | Identifier): boolean {
+            const strictlyValueMeanings = SymbolFlags.Value & ~SymbolFlags.Type;
+            const strictlyTypeMeanings = SymbolFlags.Type & ~SymbolFlags.Value;
+
+            if (!(meaning & strictlyValueMeanings) || meaning & SymbolFlags.NamespaceModule) {
+                return false;
+            }
+
+            const nameAsType = resolveName(errorLocation, name, strictlyTypeMeanings, nameNotFoundMessage, nameArg);
+            if (!nameAsType) {
+                return false;
+            }
+
+            if (nameAsType.flags & SymbolFlags.Alias) {
+                const resolvedSymbol = resolveAlias(nameAsType);
+                if (resolvedSymbol.flags & SymbolFlags.NamespaceModule) {
+                    return false;
+                }
+            }
+
+            error(errorLocation, Diagnostics.Cannot_find_name_0_A_type_exists_with_this_name_but_no_value, name);
+            return true;
+        }

```

Comment:
I got part of the way there. Wasn't able to find a way around calling resolveAlias a second time.